File: Medium/Longest_Substring_Without_Repeating_Characters/Longest_Substring_Without_Repeating_Characters.js
class Solution {
    /**
     * @param {string} s
     * @return {number}
     */
    lengthOfLongestSubstring(s) {
        let myMap  = new Map();
        let res = 0, l = 0;
        for (let i = 0; i < s.length; i++) {
            if (myMap.has(s[i])) {
                l = Math.max(myMap.get(s[i]) + 1, l);
            }

            myMap.set(s[i], i);
            res = Math.max(res, i - l + 1);
        }

        return res;
    }
}

File: Medium/Longest_Substring_Without_Repeating_Characters/Longest_Substring_Without_Repeating_Characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        Map<Character, Integer> myMap = new HashMap<>();
        int res = 0;
        for (int i = 0; i < s.length(); i++) {
            int diff = 0;
            if (myMap.containsKey(s.charAt(i))) {
                diff = i - myMap.get(s.charAt(i));
            }

            myMap.put(s.charAt(i), i);
            res = Math.max(res, diff);
        }

        return res;
    }
}


